qiskit/pulse/macros.py

```diff
@@ -104,3 +221,35 @@ def measure_all(backend) -> Schedule:
         A schedule corresponding to the inputs provided.
     """
     return measure(qubits=list(range(backend.configuration().n_qubits)), backend=backend)
+
+
+def _schedule_remapping_memory_slot(
+    schedule: Schedule, qubit_mem_slots: Dict[int, int]
+) -> Schedule:
+    """
+    A helper function to overwrite MemorySlot index of :class:`.Acquire` instruction.
+
+    Args:
+        schedule: A measurement schedule.
+        qubit_mem_slots: Mapping of measured qubit index to classical bit index.
+
+    Returns:
+        A measurement schedule with new memory slot index.
+    """
+    new_schedule = Schedule()
+    for t0, inst in schedule.instructions:
+        if isinstance(inst, instructions.Acquire):
+            qubit_index = inst.channel.index
+            reg_index = qubit_mem_slots.get(qubit_index, 0)
```

Comment:
The unittest you wrote made me think this should be 
```suggestion
            reg_index = qubit_mem_slots.get(qubit_index, qubit_index)
```
because all acquisition instruction might store results in the slot0 when you specify `qubit_mem_slots={0:0}` (i.e. you may provide the slot only for qubit of interested). This causes data conflict on the slot and you may get meaningless outcome from experiment.